File: src/test/java/net/sf/rails/game/state/ModelImpl.java
package net.sf.rails.game.state;

import net.sf.rails.game.state.Item;
import net.sf.rails.game.state.Model;
import net.sf.rails.game.state.State;
import net.sf.rails.game.state.StringState;

//An implementation only for testing
class ModelImpl extends Model {

    private final StringState text = StringState.create(this, "text");
    
    private ModelImpl(Item parent, String id, String text) {
        super(parent, id);
        this.text.set(text);
    }
    
    static ModelImpl create(Item parent, String id, String text) {
        return new ModelImpl(parent, id, text);
    }
    
    void changeText(String text) {
        this.text.set(text);
    }
    
    State getState() {
        return text;
    }
    
    @Override
    public String toText() {
        return text.value();
    }
}

File: dotnet/GameStateTest/ModelImpl.cs
using GameLib.Net.Game.State;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace GameStateTest
{
    public class ModelImpl : Model
    {
        private StringState text;


        private ModelImpl(IItem parent, string id, string text) : base(parent, id)
        {
            this.text = StringState.Create(this, "text");

            this.text.Set(text);
        }

        public static ModelImpl Create(IItem parent, string id, string text)
        {
            return new ModelImpl(parent, id, text);
        }

        public void ChangeText(string text)
        {
            this.text.Set(text);
        }

        public GameState GameState
        {
            get
            {
                return text;
            }
        }

        override public string ToText()
        {
            return text.Value;
        }
    }
}
